src/fairchem/data/oc/core/interface_config.py

```diff
@@ -0,0 +1,244 @@
+                "solvent_volume": volume,
+                "slab_millers": self.slab.millers,
+                "slab_shift": self.slab.shift,
+                "slab_top": self.slab.top,
+                "bulk_idx": self.slab.bulk.bulk_id_from_db,
+            }
+            metadata_interface_list.append(metadata)
+
+        return atoms_interface_list, metadata_interface_list
+
+    def create_packmol_atoms(self, geometry, n_solvent_mols):
+        cell = geometry.cell
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            output_path = os.path.join(tmp_dir, "out.pdb")
+            self.solvent.atoms.write(f"{tmp_dir}/solvent.pdb", format="proteindatabank")
+
+            # When placing a single ion, packmol strangely always places it at
+            # the boundary of the cell. This hacky fix manually places
+            # the ion in a random location in the cell. Packmol then will fix
+            # these atoms and not optimize them during its optimization, only
+            # optimizing solvent molecules arround the ion.
+            for i, ion in enumerate(self.ions):
+                ion_atoms = ion.atoms.copy()
+                ion_atoms.set_cell(cell)
+                self.randomize_coords(ion_atoms)
+                ion_atoms.write(f"{tmp_dir}/ion_{i}.pdb", format="proteindatabank")
+
+            # write packmol input
+            packmol_input = os.path.join(tmp_dir, "input.inp")
+            with open(packmol_input, "w") as f:
+                f.write(f"tolerance {self.packmol_tolerance}\n")
+                f.write("filetype pdb\n")
+                f.write(f"output {output_path}\n")
+
+                # write solvent
+                f.write(
+                    geometry.packmol_structure(
+                        f"{tmp_dir}/solvent.pdb", n_solvent_mols, "inside"
+                    )
+                )
+
+                for i in range(len(self.ions)):
+                    f.write(f"structure {tmp_dir}/ion_{i}.pdb\n")
+                    f.write("  number 1\n")
+                    f.write("  fixed 0 0 0 0 0 0\n")
+                    f.write("end structure\n\n")
+
+            self.run_packmol(packmol_input)
+
+            solvent_ions_atoms = ase.io.read(output_path, format="proteindatabank")
+            solvent_ions_atoms.set_pbc(True)
+            solvent_ions_atoms.set_tags([3] * len(solvent_ions_atoms))
+
+        return solvent_ions_atoms
+
+    def run_packmol(self, packmol_input):
+        packmol_cmd = which("packmol")
+        if not packmol_cmd:
+            raise OSError("packmol not found.")
+
+        ps = subprocess.Popen(
+            f"{packmol_cmd} < {packmol_input}",
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        out, err = ps.communicate()
+        if err:
+            raise OSError(err.decode("utf-8"))
+
+    def randomize_coords(self, atoms):
+        cell_weights = np.random.rand(3)
```

Comment:
I recommend using random functions from numpy via a generator, with `rng = np.random.default_rng(seed=...)`. I think this is the recommended way to do it now, and gives us more control to explicitly set the seed for reproducibility.

see: https://numpy.org/doc/stable/reference/random/index.html